patches/api/0008-Paper-Plugins.patch

```diff
@@ -0,0 +1,2486 @@
++                } catch (ClassCastException ex) {
++                    throw new ClassCastException("class '%s' does not extend '%s'".formatted(clazz, classType));
++                }
++
++                clazzInstance = pluginClass.getDeclaredConstructor().newInstance();
++            } catch (IllegalAccessException exception) {
++                throw new RuntimeException("No public constructor");
++            } catch (InstantiationException exception) {
++                throw new RuntimeException("Abnormal class instantiation", exception);
++            }
++
++            return clazzInstance;
++        } catch (Throwable e) {
++            if (onError != null) {
++                onError.run();
++            }
++            SneakyThrow.sneaky(e);
++        }
++
++        throw new AssertionError(); // Shouldn't happen
++    }
++
++}
+diff --git a/src/main/java/io/papermc/paper/registry/RegistryAccess.java b/src/main/java/io/papermc/paper/registry/RegistryAccess.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..ce66c664003e80087df88bded37ef17bd398bf85
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/registry/RegistryAccess.java
+@@ -0,0 +1,16 @@
++package io.papermc.paper.registry;
++
++import org.jetbrains.annotations.NotNull;
++
++public class RegistryAccess {
++
++    /**
++     * Stub
++     *
++     * @return
++     */
++    @NotNull
++    public static RegistryAccess getInstance() {
++        return new RegistryAccess();
++    }
++}
+diff --git a/src/main/java/org/bukkit/command/PluginCommand.java b/src/main/java/org/bukkit/command/PluginCommand.java
+index 1dbbc244309043b18c1d71707c4fb066c0d0e02d..08d3ff24d5318b0c77d6467bcb705bffac202842 100644
+--- a/src/main/java/org/bukkit/command/PluginCommand.java
++++ b/src/main/java/org/bukkit/command/PluginCommand.java
+@@ -14,7 +14,7 @@ public final class PluginCommand extends Command implements PluginIdentifiableCo
+     private CommandExecutor executor;
+     private TabCompleter completer;
+ 
+-    protected PluginCommand(@NotNull String name, @NotNull Plugin owner) {
++    public PluginCommand(@NotNull String name, @NotNull Plugin owner) { // Paper
+         super(name);
+         this.executor = owner;
+         this.owningPlugin = owner;
+diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
+index e195e74c48c69047aa825b75fad95419c505b41f..53f28c9e6843991486a576d41b6641c170589807 100644
+--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
++++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
+@@ -34,7 +34,7 @@ public class SimpleCommandMap implements CommandMap {
+     private void setDefaultCommands() {
+         register("bukkit", new VersionCommand("version"));
+         register("bukkit", new ReloadCommand("reload"));
+-        register("bukkit", new PluginsCommand("plugins"));
++        //register("bukkit", new PluginsCommand("plugins")); // Paper
```

Comment:
clarify why this happens, probably by deleting the line sayings its been replaced on the server.